realm/realm-library/src/androidTest/java/io/realm/MutableRealmIntegerTests.java

```diff
@@ -366,12 +363,12 @@ private void testNullability(MutableRealmIntegerTypes c1) {
             c1.columnMutableRealmInteger.increment(5);
             fail("Attempt to increment a null valued MutableRealmInteger should throw NPE");
         }
-        catch (NullPointerException ignore) { }
+        catch (IllegalArgumentException ignore) { }
```

Comment:
Well, close.  NPE would be the most accurate.  To do that, though, I'd have to catch the IAE thrown by the native code, check the message, and throw the NPE.  I even think that might be the right thing to do.